examples/tensorflow/segmentation/evaluation.py

```diff
@@ -184,10 +185,21 @@ def run_evaluation(config, eval_timeout=None):
                                 weights=config.get('weights', None),
                                 is_training=False) as model:
         with strategy.scope():
-            compression_ctrl, compress_model = create_compressed_model(model, config.nncf_config)
+            compression_state = TFCompressionState()
+            checkpoint = tf.train.Checkpoint(compression_state=compression_state)
+            if 'test' in config.mode and config.ckpt_path:
+                load_checkpoint(checkpoint, config.ckpt_path)
+            elif 'train' in config.mode:
+                checkpoint_dir = config.checkpoint_save_dir
+                eval_timeout = config.eval_timeout
+                for checkpoint_path in tf.train.checkpoints_iterator(checkpoint_dir, timeout=eval_timeout):
+                    checkpoint.restore(checkpoint_path)
```

Comment:
I hoped it will be much shorter for segmentation sample, but can't find another way of loading compression state for all cases
@alexsu52